include/bredis/impl/connection.ipp

```diff
@@ -174,4 +150,19 @@ Connection<NextLayer>::read(DynamicBuffer &rx_buff,
     return boost::get<positive_parse_result_t>(parse_result);
 }
 
+template <typename NextLayer>
+template <typename DynamicBuffer>
+inline positive_parse_result_t
+Connection<NextLayer>::read(DynamicBuffer &rx_buff) {
+    namespace asio = boost::asio;
+    namespace sys = boost::system;
+
+    boost::system::error_code ec;
+    auto result = read(rx_buff, ec);
```

Comment:
I'm not 100% sure but I _think_ you might need to write
```
auto result = this->read(rx_buff, ec);
```

Otherwise if a free function `read` which takes the concrete type of `DynamicBuffer` exists, it might be selected instead of the member function. I'm not really sure, you should ask around or check the standard.